queue_processors/queue_processor/tests/test_queue_listener_daemon.py

```diff
@@ -0,0 +1,116 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Tests the Queue Listener Daemon
+"""
+
+import unittest
+from unittest import mock
+
+from queue_processors.queue_processor.queue_listener_daemon import \
+    QueueListenerDaemon
+import queue_processors.queue_processor.queue_listener_daemon
```

Comment:
Do we need both of these imports or could we just use one?

Reply:
This is a subtlety with patching that I really dislike and run into all the time:

If our module is `foo.bar`

`from x import y` we now need to patch `foo.bar.y`
`import x` we now patch `x.y`

As we are patching an import happening in module `x` rather that `foo` it makes it much easier to just do a bare import rather than a from import then try and decipher the new namespace.

For this reason I've left it as a `from` and `import`, any other time agree completely